synapseclient/table.py

```diff
@@ -829,75 +857,155 @@ class Dataset(SchemaBase):
     :py:classmethod:`synapseclient.client.create_snapshot_version`.
 
     Example::
+
         syn = synapseclient.login()
         syn.create_snapshot_version(
             dataset.id,
             label="v1.0",
             comment="This is version 1")
     """
     _synapse_entity_type: str = "org.sagebionetworks.repo.model.table.Dataset"
-    _property_keys: List[str] = SchemaBase._property_keys + ['dataset_items']
+    _property_keys: List[str] = SchemaBase._property_keys + ['datasetItems']
+    _local_keys: List[str] = SchemaBase._local_keys + ['folders_to_add', 'force']
 
     def __init__(self, name=None, columns=None, parent=None, properties=None,
-                 annotations=None, local_state=None, dataset_items=None, **kwargs):
-        self.properties.setdefault('dataset_items', [])
+                 annotations=None, local_state=None, dataset_items=None,
+                 folder_ids=None, force=None, **kwargs):
+        self.properties.setdefault('datasetItems', [])
+        self.__dict__.setdefault('folders_to_add', set())
+        self.__dict__.setdefault('force', False)
         super(Dataset, self).__init__(
             name=name, columns=columns, properties=properties,
             annotations=annotations, local_state=local_state, parent=parent,
             **kwargs
         )
+
+        # Add default Dataset columns if schema is not provided.
+        if not self.properties.columnIds and columns is None:
+            self.addColumns(['81721', '81722', '81723', '81724', '81725',
+                             '81728', '81729', '81730', '81731', '81732',
+                             '81726', '81727', '81733', '112368', '112369'])
+        if force:
+            self.force = True
         if dataset_items:
-            self.add_items(dataset_items)
+            self.add_items(dataset_items, force)
+        if folder_ids:
+            self.add_folders(folder_ids, force)
 
     def __len__(self):
-        return len(self.properties.dataset_items)
+        return len(self.properties.datasetItems)
+
+    def _check_needed_keys(syn, keys: List[str]):
```

Comment:
This seems like it can be a convenience function separate from the class. Why keep it in the class?

Reply:
Interesting. So pass necessary keys as a param, then?

What other classes would need this check?

Reply:
I was referring to the fact that `_check_needed_keys(` didn't have a `self` as a parameter which makes it a little awkward in the class.  But, you're right, no other classes need this check.

Reply:
Ah, you are right (also - why did I pass `syn`....? ;;;).  Thanks for pointing it out - I'll make a note that this is a static method.